app/controllers/concerns/client_access_control_concern.rb

```diff
@@ -4,9 +4,13 @@ module ClientAccessControlConcern
   private
 
   def require_client_login
-    unless current_client.present?
+    if current_client.blank?
       session[:after_client_login_path] = request.original_fullpath if request.get?
-      redirect_to new_portal_client_login_path
+      redirect_to new_portal_client_login_path and return
+    end
+
+    if StillNeedsHelpService.must_show_still_needs_help_flow?(current_client)
```

Comment:
See previous PR suggestion to break this latter part of the method into its own method. You could update portal controller to chain the before_actions :require_client_login, :redirect_if_still_needs_help